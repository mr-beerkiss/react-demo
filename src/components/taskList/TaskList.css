.wrapper {
  width: 100%;
  max-width: 500px;
  margin: 0 auto;
}

.tasks {
}

.newTask {
  font-size: inherit;
  margin-top: 5px;
}
