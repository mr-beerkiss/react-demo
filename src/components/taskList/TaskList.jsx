import React, {Component, PropTypes} from 'react'
import styles from './TaskList.css'
import TaskListItem from './TaskListItem.jsx'

export default class TaskList extends Component {
  static propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      complete: PropTypes.bool.isRequired,
      archived: PropTypes.bool.isRequired
    })),
    onTaskChange: PropTypes.func,
    onTaskComplete: PropTypes.func,
    onTaskArchive: PropTypes.func,
    onNewTask: PropTypes.func
  }

  static defaultProps = {
    data: []
  }

  render () {
    const {
      data,
      onTaskChange,
      onTaskComplete,
      onTaskArchive,
      onNewTask
    } = this.props

    return (
      <div className={styles.wrapper}>
        <div className={styles.tasks}>
          {data.map((item, index) => (
            <TaskListItem
              key={index}
              name={item.name}
              complete={item.complete}
              archived={item.archived}
              onChange={(value) => onTaskChange && onTaskChange(index, value)}
              onComplete={(complete) => onTaskComplete && onTaskComplete(index, complete)}
              onArchive={(archive) => onTaskArchive && onTaskArchive(index, archive)} />
          ))}
        </div>
        <button
          className={styles.newTask}
          onClick={() => onNewTask && onNewTask()}>
          New Task
        </button>
      </div>
    )
  }
}
