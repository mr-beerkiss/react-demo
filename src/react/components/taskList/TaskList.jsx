import React, {Component, PropTypes} from 'react'
import styles from './TaskList.css'
import TaskListItem from './TaskListItem.jsx'

export default class TaskList extends Component {
  static propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      complete: PropTypes.bool.isRequired,
      removed: PropTypes.bool.isRequired
    })),
    onTaskChange: PropTypes.func,
    onTaskComplete: PropTypes.func,
    onTaskRemove: PropTypes.func,
    onAddTask: PropTypes.func
  }

  static defaultProps = {
    data: []
  }

  render () {
    const {
      data,
      onTaskChange,
      onTaskComplete,
      onTaskRemove,
      onAddTask
    } = this.props

    return (
      <div className={styles.wrapper}>
        <div className={styles.tasks}>
          {data.map((item, index) => (
            <TaskListItem
              key={index}
              name={item.name}
              complete={item.complete}
              removed={item.removed}
              onChange={(value) => onTaskChange && onTaskChange(index, value)}
              onComplete={(complete) => onTaskComplete && onTaskComplete(index, complete)}
              onRemove={() => onTaskRemove && onTaskRemove(index)} />
          ))}
        </div>
        <button
          className={styles.newTask}
          onClick={() => onAddTask && onAddTask()}>
          Add Task
        </button>
      </div>
    )
  }
}
