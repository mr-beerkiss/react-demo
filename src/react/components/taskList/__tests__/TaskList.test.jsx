/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'
import TaskList from '../TaskList.jsx'
import styles from '../TaskList.css'

describe('<TaskList />', () => {
  it('snapshot: render header and title', () => {
    const wrapper = shallow(
      <TaskList />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('new task onClick handler triggers', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(
      <TaskList onAddTask={mockFn} />
    )

    const button = wrapper.find(`.${styles.newTask}`)
    button.simulate('click')
    expect(mockFn.mock.calls[0]).toEqual([])
  })
})
