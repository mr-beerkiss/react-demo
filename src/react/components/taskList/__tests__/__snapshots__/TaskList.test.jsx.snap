exports[`<TaskList /> snapshot: render header and title 1`] = `
<div
  className="wrapper">
  <div
    className="tasks" />
  <button
    className="newTask"
    onClick={[Function]}>
    Add Task
  </button>
</div>
`;
